CRM/Activity/Form/Activity.php

```diff
@@ -265,33 +268,20 @@ public function preProcess() {
 
     $this->assign('context', $this->_context);
 
-    $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this);
+    $this->setAction(CRM_Utils_Request::retrieve('action', 'String', $this));
```

Comment:
Once https://github.com/civicrm/civicrm-core/pull/27495 is merged it will cast to an int & we can be confident it is numeric